lib/hackney/notification/send_manual_precompiled_letter.rb

```diff
@@ -11,7 +11,7 @@ def execute(username: nil, payment_ref: nil, template_id:, unique_reference:, le
         # FIXME: this must be in a background job => UH is unreliable
         # TODO: create job to accept exact same args as add_action_diary_usecase
 
-        tenancy_ref = leasehold_gateway.get_tenancy_ref(payment_ref: payment_ref).dig(:tenancy_ref)
+        tenancy_ref = leasehold_gateway.get_tenancy_ref(payment_ref: payment_ref).dig(:tenancy_ref) if tenancy_ref.nil?
```

Comment:
```suggestion
        tenancy_ref ||= leasehold_gateway.get_tenancy_ref(payment_ref: payment_ref).dig(:tenancy_ref)
```

More idiomatic ruby to me.

Reply:
I forgot we can do this :)